src/daemon/daemon.cpp

```diff
@@ -2024,3 +2043,38 @@ grpc::Status mp::Daemon::cmd_vms(const std::vector<std::string>& tgts, std::func
 
     return grpc::Status::OK;
 }
+
+void mp::Daemon::install_sshfs(const VirtualMachine::UPtr& vm, const std::string& name)
+{
+    auto& key_provider = *config->ssh_key_provider;
+
+    SSHSession session{vm->ssh_hostname(), vm->ssh_port(), vm->ssh_username(), key_provider};
+
+    mpl::log(mpl::Level::info, category, fmt::format("Installing sshfs in \'{}\'", name));
+
+    int retries{0};
+    while (++retries <= 3)
```

Comment:
3 should be a constant somewhere IMO. Either at the top of this file, if only used here, or globally available if we do this sort of retrying elsewhere. There is something [here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es45-avoid-magic-constants-use-symbolic-constants), but I think the main argument is encapsulation (e.g. to avoid changing this 3 in the future but missing the one below)

Reply:
Agreed...I must have had a moment of insanity when I did this.  I know better:)

Reply:
Done